airbyte-integrations/connectors/source-bigquery/source_bigquery/source.py

```diff
@@ -52,7 +54,25 @@ def check_connection(self, logger, config: Mapping[str, Any]) -> Tuple[bool, any
         :param logger:  logger object
         :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
         """
-        return False, ""
+        try:
+            self._auth = BigqueryAuth(config)
+            # try reading first table from each dataset, to check the connectivity,
+            for dataset in BigqueryDatasets(project_id=config["project_id"], authenticator=self._auth).read_records(sync_mode=SyncMode.full_refresh):
```

Comment:
This check currently validates the following:
- It checks that the credentials provided by the user are valid, otherwise we will have an error
- It checks that the user has access to at least one dataset and at least one table in each dataset. If the user does not have access to at least one dataset the check will fail with the error `'NoneType' object is not iterable"` , but I think this error is not telling of the source of the problem so I will update it to give a more relevant error message

I see for the pushdown filters (I didn't think about this), I will add this in the check too then.

Reply:
Can you point out where it raises `'NoneType' object is not iterable"`? This should be explicit if possible.

Can you also show me the path where `check_connection` will return `False`? This should happen if credentials are invalid